src/odbc.cpp

```diff
@@ -242,31 +247,28 @@ ODBCError* ODBCAsyncWorker::GetODBCErrors
 
     ODBCError error;
 
-
-    return_code = SQLGetDiagRec(
+    return_code = SQLGetDiagRec
+    (
       handleType,                 // HandleType
       handle,                     // Handle
       i + 1,                      // RecNumber
-      errorSQLState,              // SQLState
-      &nativeError,               // NativeErrorPtr
+      error.state,                // SQLState
+      &error.code,                // NativeErrorPtr
       errorMessage,               // MessageText
       ERROR_MESSAGE_BUFFER_CHARS, // BufferLength
       &textLength                 // TextLengthPtr
     );
 
     if (SQL_SUCCEEDED(return_code)) {
-      error.state = errorSQLState;
-      error.code = nativeError;
       #ifdef UNICODE
         byteCount = (std::char_traits<char16_t>::length((char16_t *)errorMessage) + 1) * 2;
       #else
         byteCount = std::strlen((const char *)errorMessage) + 1;
       #endif
       error.message = new SQLTCHAR[byteCount];
```

Comment:
Why are we allocating an array based off a byte count? Arrays are based element count, not byte lengths.

Reply:
byteCount is calculated for the memcpy below here

Reply:
Yes, but `new SQLTCHAR[byteCount];` should not be using byteCount

Reply:
True, I changed the logic to use `error_message_length`, which returns `the total number of characters (excluding the number of characters required for the null-termination character) available to return in *MessageText.` This also allowed me to eliminate some of the `ifdef`s